consensus/ibft/src/integration-test/java/tech/pegasys/pantheon/consensus/ibft/tests/GossipTest.java

```diff
@@ -73,59 +68,56 @@
   private Block block;
   private ValidatorPeer gossiper;
   private MessageFactory msgFactory;
-  private SignedData<ProposalPayload> proposal;
 
   @Before
   public void setup() {
     context.getController().start();
     block = context.createBlockForProposalFromChainHead(roundId.getRoundNumber(), 30);
     gossiper = roles.getProposer();
     msgFactory = gossiper.getMessageFactory();
-
-    proposal = msgFactory.createSignedProposalPayload(roundId, block);
   }
 
   @Test
   public void gossipMessagesToPeers() {
-    gossiper.injectProposal(roundId, block);
-    assertPeersContainsReceivedMessages(roles.getNonProposingPeers(), proposal);
+    SignedData<PreparePayload> localPrepare =
+        context.getLocalNodeMessageFactory().createSignedPreparePayload(roundId, block.getHash());
+    assertPeersReceivedNoMessages(roles.getNonProposingPeers());
+    final SignedData<ProposalPayload> proposal = gossiper.injectProposal(roundId, block);
+    // Gossiper node will have a prepare message as an effect of the proposal being sent
+    assertPeersReceivedExactly(roles.getNonProposingPeers(), proposal, localPrepare);
+    assertPeersReceivedExactly(singleton(gossiper), localPrepare);
 
-    gossiper.injectPrepare(roundId, block.getHash());
-    final SignedData<PreparePayload> prepare =
-        msgFactory.createSignedPreparePayload(roundId, block.getHash());
-    assertPeersContainsReceivedMessages(roles.getNonProposingPeers(), prepare);
-
-    gossiper.injectCommit(roundId, block.getHash());
-    final IbftExtraData extraData = IbftExtraData.decode(block.getHeader().getExtraData());
-    final Hash commitHash =
-        IbftBlockHashing.calculateDataHashForCommittedSeal(block.getHeader(), extraData);
-    final Signature commitSeal = SECP256K1.sign(commitHash, gossiper.getNodeKeyPair());
-    final SignedData<CommitPayload> commit =
-        msgFactory.createSignedCommitPayload(roundId, block.getHash(), commitSeal);
-    assertPeersContainsReceivedMessages(roles.getNonProposingPeers(), commit);
+    final SignedData<PreparePayload> prepare = gossiper.injectPrepare(roundId, block.getHash());
```

Comment:
technically the gossiper SHOULDN'T send in the prepare (proposer can't also prepare) ... but not too worried.